Sources/ComposedUI/CollectionView/Handlers/CollectionTapSelectionHandler.swift

```diff
@@ -0,0 +1,22 @@
+import Composed
+import UIKit
+
+/// A type that handles taps on individual cells.
+///
+/// When conforming to this protocol do not implement any of the methods from
+/// ``CollectionSelectionHandler``; only implement ``didTapElement(at:cell:)`` or
+/// `didTapElement(at:)` and handle the tap there.
+public protocol CollectionTapSelectionHandler: CollectionSelectionHandler, TapSelectionHandler {
```

Comment:
In my view, the `CollectionTapSelectionHandler`, being a type of `SelectionHandler`, introduces a side effect within `shouldSelectCell(_:at:)`. This could lead to potential issues if `shouldSelectCell(_:at:)` is overridden by subclasses.

I believe the underlying issue might stem from the `allowsMultipleSelection` property being set to `true` for all instances within `ComposedUI`. Could this be the case?

Reply:
> In my view, the `CollectionTapSelectionHandler`, being a type of `SelectionHandler`, introduces a side effect within `shouldSelectCell(_:at:)`. This could lead to potential issues if `shouldSelectCell(_:at:)` is overridden by subclasses.

I agree this isn't ideal. For some context I did this because the idea behind `ComposedUI` is that it exposes `UICollectionViewDelegate` functions to each `Section`. This would be a diversion from this, which I would hope to avoid.

I thought about adding a custom SwiftLint rule for this. I have not looked in to it enough to know if we could include this in Composed or if it would need to be in the main app repo.

> I believe the underlying issue might stem from the `allowsMultipleSelection` property being set to `true` for all instances within `ComposedUI`. Could this be the case?

Even if `allowsMultipleSelection` were `false` (either as a default or we set it ourselves) it would still keep the selection until a different cell is selected or the selected cell is tapped again. Being `true` can make this worse because a cell is never automatically deselected when using `didSelect(at:)`, but being `false` isn't going to entirely fix this.